src/simulation/dynamics/_GeneralModuleFiles/svIntegratorRungeKutta.h

```diff
@@ -0,0 +1,208 @@
+/*
+ ISC License
+
+ Copyright (c) 2021, Autonomous Vehicle Systems Lab, University of Colorado at Boulder
+
+ Permission to use, copy, modify, and/or distribute this software for any
+ purpose with or without fee is hereby granted, provided that the above
+ copyright notice and this permission notice appear in all copies.
+
+ THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+ */
+
+#ifndef svIntegratorRungeKutta_h
+#define svIntegratorRungeKutta_h
+
+#include "../_GeneralModuleFiles/dynamicObject.h"
+#include "../_GeneralModuleFiles/dynParamManager.h"
+#include "../_GeneralModuleFiles/stateVecIntegrator.h"
+#include "extendedStateVector.h"
+#include <array>
+#include <functional>
+#include <memory>
+#include <stdint.h>
+#include <unordered_map>
+
+#include <iostream>
+
+/**
+ * Stores the coefficients necessary to use the Runge-Kutta methods.
+ *
+ * The Butcher table looks like:
+ *
+ *   c | A
+ *     -----
+ *       b
+ */
+template <size_t numberStages> struct RKCoefficients {
+
+    /** Array with size = numberStages */
+    using StageSizedArray = std::array<double, numberStages>;
+
+    /** Square matrix with size = numberStages */
+    using StageSizedMatrix = std::array<StageSizedArray, numberStages>;
+
+    // = {} performs zero-initialization
+    StageSizedMatrix aMatrix = {}; /**< "a" coefficient matrix of the RK method */
+    StageSizedArray bArray = {};   /**< "b" coefficient array of the RK method */
+    StageSizedArray cArray = {};   /**< "c" coefficient array of the RK method */
+};
+
+/**
+ * The svIntegratorRungeKutta class implements a state integrator based on the
+ * family of explicit Runge-Kutta numerical integrators.
+ *
+ * A Runge-Kutta method is defined by its stage number and its coefficients.
+ * The stage number drives the computational cost of the method: an RK method
+ * of stage 4 requires 4 dynamics evaluations (FSAL optimizations are not done).
+ *
+ * Note that the order of the integrator is lower or equal to the stage number.
+ * A RK method of order 5, for example, requires 7 stages.
+ */
+template <size_t numberStages> class svIntegratorRungeKutta : public StateVecIntegrator {
+  public:
+    static_assert(numberStages > 0, "One cannot declare Runge Kutta integrators of stage 0");
```

Comment:
The `static_assert` happens at compile time, which means that it's impossible to compile code for a `svIntegratorRungeKutta` with zero stages, as the algorithm is not defined for this case. I'd argue that the earlier we throw errors the better, so better to keep the error at compile time rather than runtime, id say.